ext/semian/semian.h

```diff
@@ -0,0 +1,49 @@
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/sem.h>
+#include <sys/time.h>
+#include <errno.h>
+#include <string.h>
+
+#include <ruby.h>
+#include <ruby/util.h>
+#include <ruby/io.h>
+
+#include <openssl/sha.h>
+
+#include <stdio.h>
+
+#include <sys/shm.h>
+#include <unistd.h>
+
+#include <math.h>
+
+union semun {
+  int              val;    /* Value for SETVAL */
+  struct semid_ds *buf;    /* Buffer for IPC_STAT, IPC_SET */
+  unsigned short  *array;  /* Array for GETALL, SETALL */
+  struct seminfo  *__buf;  /* Buffer for IPC_INFO
+                             (Linux-specific) */
+};
+
+#if defined(HAVE_RB_THREAD_CALL_WITHOUT_GVL) && defined(HAVE_RUBY_THREAD_H)
+// 2.0
+#include <ruby/thread.h>
+#define WITHOUT_GVL(fn,a,ubf,b) rb_thread_call_without_gvl((fn),(a),(ubf),(b))
+#elif defined(HAVE_RB_THREAD_BLOCKING_REGION)
+ // 1.9
+typedef VALUE (*my_blocking_fn_t)(void*);
+#define WITHOUT_GVL(fn,a,ubf,b) rb_thread_blocking_region((my_blocking_fn_t)(fn),(a),(ubf),(b))
+#endif
+
+VALUE eSyscall, eTimeout, eInternal;
+
+key_t
+generate_key(const char *name);
+
+void
+raise_semian_syscall_error(const char *syscall, int error_num);
+
+
+void
+set_semaphore_permissions(int sem_id, int permissions);
```

Comment:
The `<type>\n<fname>` idiom is not used [for header files in Ruby](https://github.com/ruby/ruby/blob/trunk/internal.h#L747). Just keep them on the same line.